src/Tring/Program.cs

```diff
@@ -29,73 +33,124 @@ static int Main(string[] args)
+                        await SetupConnections(optionWatch.Value() == "on", arguments.Values);
+                        OutputPrinter.CleanUp(arguments.Values.Count);
+                        break;
                 }
-                Connect(optionWatch, connectionTester);
                 return 0;
             });
             try
             {
                 return app.Execute(args);
             }
-            catch (Exception e)
+            catch (AggregateException exception)
+            {
+                var errors = exception.InnerExceptions.Where(e => !(e is TaskCanceledException));
+                foreach(var e in errors)
+                {
+                    Console.ForegroundColor = ConsoleColor.Red;
+                    Console.Error.WriteLine(e.Message);
+                    Console.ResetColor();
+                }
+                return errors.Any() ? -1 : 0; 
+            }
+            catch (Exception exception) when (!(exception is TaskCanceledException))
             {
                 Console.ForegroundColor = ConsoleColor.Red;
-                Console.Error.WriteLine(e.Message);
+                Console.Error.WriteLine(exception.Message);
                 Console.ResetColor();
                 return -1;
             }
         }
 
-        private static void Connect(CommandOption optionWatch, ConnectionTester connectionTester)
+        private static Task SetupConnections(bool watchMode, List<string> connections)
         {
-            OutputPrinter.SetupCleanUp();
-            OutputPrinter.PrintTable();
-            var startTime = DateTimeOffset.Now;
-            ConnectionResult result = null;
-            while (true)
+            var tasks = new List<Task>();
+            var connectors = new List<ConnectionTester>();
+            foreach (string input in connections)
             {
-                var watch = System.Diagnostics.Stopwatch.StartNew();
-                var newResult = connectionTester.TryConnect();
-                if (result?.IsEquivalent(newResult) ?? false)
-                {
-                    if (!Console.IsOutputRedirected)
-                        Console.CursorTop--;
-                }
-                else
-                {
-                    result = newResult;
-                    startTime = newResult.TimeStamp;
-                }
-                OutputPrinter.ResetPrintLine();
-                OutputPrinter.PrintLogEntry(startTime, newResult);
-                if (optionWatch.Value() != "on")
-                    break;
-                OutputPrinter.HideCursor();
-                if (Console.KeyAvailable && Console.ReadKey().Key == ConsoleKey.Escape)
+                connectors.Add(new ConnectionTester(input));
+            }
+            var ipSize = connectors.Select(connection => connection.request.Ip).Max(ip => ip?.ToString()?.Length) ?? 15;
```

Comment:
I seems odd to me that the `ConnectionTester` has to expose the request. Probably it is nice if the input is first mapped to a list of requests which can be mapped to a list of `ConnectionTester` objects. Then the calculation for the max ip size and port size can be done using the list of requests.